neuralop/layers/segment_csr.py

```diff
@@ -3,50 +3,79 @@
 
 import torch
 
-def segment_csr(src: torch.Tensor, indptr: torch.Tensor, reduce: Literal['mean', 'sum'], use_scatter=True):
-    """segment_csr reduces all entries of a CSR-formatted 
-    matrix by summing or averaging over neighbors. 
 
-    Used to reduce features over neighborhoods 
+def segment_csr(
+    src: torch.Tensor,
+    indptr: torch.Tensor,
+    reduce: Literal["mean", "sum"],
+    use_scatter=True,
+):
+    """segment_csr reduces all entries of a CSR-formatted
+    matrix by summing or averaging over neighbors.
+
+    Used to reduce features over neighborhoods
     in neuralop.layers.IntegralTransform
-    
+
     Parameters
     ----------
     src : torch.Tensor
         tensor of features for each point
     indptr : torch.Tensor
-        splits representing start and end indices 
+        splits representing start and end indices
         of each neighborhood in src
     reduce : Literal['mean', 'sum']
         how to reduce a neighborhood. if mean,
         reduce by taking the average of all neighbors.
-        Otherwise take the sum. 
+        Otherwise take the sum.
     """
-    if reduce not in ['mean', 'sum']:
-        raise ValueError("reduce must be one of \'mean\', \'sum\'")
-    
-    if torch.backends.cuda.is_built() and importlib.find_loader('torch_scatter') and use_scatter:
+    if reduce not in ["mean", "sum"]:
+        raise ValueError("reduce must be one of 'mean', 'sum'")
+
+    if (
+        torch.backends.cuda.is_built()
+        and importlib.util.find_spec("torch_scatter") is not None
+        and use_scatter
+    ):
         """only import torch_scatter when cuda is available"""
         import torch_scatter.segment_csr as scatter_segment_csr
+
         return scatter_segment_csr(src, indptr, reduce=reduce)
 
     else:
-        n_nbrs = indptr[1:] - indptr[:-1] # end indices - start indices
+        # if batched, shape [b, n_reps, channels]
```

Comment:
Should we add a comment that this is a naive implementation of segment_csr? Also, should we warn the user in case `use_scatter=True` but the above `if` fails?

Reply:
Thanks! Are we not using the `logging` module so users can suppress the warning?

Reply:
Agree! We are also calling the logging module in `mpu/comm.py`